### PR TITLE
Fixed PHP error: Allowed memory size ... exhausted...

### PR DESCRIPTION
**Pull request by psat01**

### CODE REVIEW

File: Decorator/ContentProviderDetachDecorator.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        $this->provider = $provider;
+        $this->manager = $manager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getReferenced($id)
+    {
+        return new DetachIterator($this->provider->getReferenced($id), $this->manager);
```
Comment (joenivl) [inline Decorator/ContentProviderDetachDecorator.php:48]: ContentProviderInterface does not return an Iterator according to the docblock. The best way to fix is is to change the docblock (change Traversable into Iterator) of the ContentProviderInterface (which is in the library)

File: Decorator/ContentTypeProviderDetachDecorator.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        $this->provider = $provider;
+        $this->manager = $manager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getContent($id)
+    {
+        return new DetachIterator($this->provider->getContent($id), $this->manager);
```
Comment (joenivl) [inline Decorator/ContentTypeProviderDetachDecorator.php:48]: ContentTypeProviderInterface does not return an Iterator according to the docblock. The best way to fix is is to change the docblock (change Traversable into Iterator) of the ContentTypeProviderInterface (which is in the library)

File: Decorator/ContentProviderDetachDecorator.php (hunk 1 above)
Comment (psat01) [reply Decorator/ContentProviderDetachDecorator.php:48]: I've changed the docblock in the library and created a new pull-request.

File: Decorator/ContentTypeProviderDetachDecorator.php (hunk 2 above)
Comment (psat01) [reply Decorator/ContentTypeProviderDetachDecorator.php:48]: I've changed the docblock in the library and created a new pull-request.

**Pull request merged**